internal/pkg/cache/redis/redis.go

```diff
@@ -39,22 +42,42 @@ func NewRedis(address string) *Redis {
 
 // NewRedisWithConfig creates a new Redis instance with configurable options.
 func NewRedisWithConfig(cfg *config.RedisConfig) *Redis {
-	o := &redis.Options{
-		Addr:         cfg.Address,
-		MinIdleConns: cfg.MinIdleConns,
-		PoolSize:     cfg.PoolSize,
-		IdleTimeout:  cfg.IdleTimeout,
-		MaxConnAge:   cfg.MaxConnAge,
+	o := &redis.UniversalOptions{
+		Addrs:           parseRedisAddress(cfg.Address),
+		MinIdleConns:    cfg.MinIdleConns,
+		PoolSize:        cfg.PoolSize,
+		ConnMaxIdleTime: cfg.IdleTimeout,
+		ConnMaxLifetime: cfg.MaxConnAge,
+	}
+
+	c := redis.NewUniversalClient(o)
+
+	err := redisotel.InstrumentMetrics(c)
+	if err != nil {
+		log.Fatalf("got error adding metric instrumentation to redis client: %v", err)
 	}
 
-	c := redis.NewClient(o)
-	c.AddHook(redisotel.NewTracingHook())
+	err = redisotel.InstrumentTracing(c)
+	if err != nil {
+		log.Fatalf("got error adding tracing instrumentation to redis client: %v", err)
```

Comment:
I would prefer to log this as an error to not fail the service at start up in case there is an issue with OTel when we release a hotfix to the service and cannot wait for troubleshooting / solving OTel beforehand.

Reply:
Fixed.